osu.Game/Screens/OnlinePlay/Components/StatusColouredContainer.cs

```diff
@@ -1,39 +1,52 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
-#nullable disable
-
+using System.ComponentModel;
 using osu.Framework.Allocation;
-using osu.Framework.Bindables;
 using osu.Framework.Graphics;
-using osu.Framework.Graphics.Containers;
 using osu.Game.Graphics;
 using osu.Game.Online.Rooms;
+using Container = osu.Framework.Graphics.Containers.Container;
 
 namespace osu.Game.Screens.OnlinePlay.Components
 {
     public partial class StatusColouredContainer : Container
     {
+        [Resolved]
+        private OsuColour colours { get; set; } = null!;
+
         private readonly double transitionDuration;
+        private readonly Room room;
+
+        public StatusColouredContainer(Room room, double transitionDuration = 100)
+        {
+            this.room = room;
+            this.transitionDuration = transitionDuration;
+        }
 
-        [Resolved(typeof(Room), nameof(Room.Status))]
-        private Bindable<RoomStatus> status { get; set; }
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
 
-        [Resolved(typeof(Room), nameof(Room.Category))]
-        private Bindable<RoomCategory> category { get; set; }
+            room.PropertyChanged += onRoomPropertyChanged;
+            updateRoomStatus();
+        }
 
-        public StatusColouredContainer(double transitionDuration = 100)
+        private void onRoomPropertyChanged(object? sender, PropertyChangedEventArgs e)
         {
-            this.transitionDuration = transitionDuration;
+            if (e.PropertyName == nameof(Room.Status))
+                updateRoomStatus();
```

Comment:
`Room.Category` changes should also presumably be handled, looking at old code, but aren't...?

This sort of thing is why I'm not the biggest fan of `INotifyPropertyChanged` from a usability angle - with the old setup, forgetting this sort of thing was extremely unlikely as long as you resolved the piece of data. Now it is easier to forget to check a property.

Reply:
As far as I can tell, the previous code did not handle changes to category. I didn't intend to make such changes/fixes here.

Reply:
Oh, right, the old code resolved a thing but never actually used it... My bad